apiserver/facades/client/usermanager/usermanager.go

```diff
@@ -383,3 +383,43 @@ func (api *UserManagerAPI) setPassword(arg params.EntityPassword) error {
 	}
 	return nil
 }
+
+// ResetPassword resets password for supplied users by
+// invalidating current passwords (if any) and generating
+// new random secret keys which will be returned.
+func (api *UserManagerAPI) ResetPassword(args params.Entities) (params.AddUserResults, error) {
+	var result params.AddUserResults
+
+	if err := api.check.ChangeAllowed(); err != nil {
+		return result, errors.Trace(err)
+	}
+
+	if len(args.Entities) == 0 {
+		return result, nil
+	}
+
+	isSuperUser, err := api.hasControllerAdminAccess()
```

Comment:
Our connections authenticate once at the beginning. Its arguably a known hole that if you delete a user/change your password/etc existing connections don't notice and don't ask you to reconnect.
So it doesn't *do* anything. If we did have appropriate logic to trigger a disconnect of all users currently connected with a credential when their credential changed, we'd have to make sure that the user could get the secret-key back before getting dropped. Otherwise the act of resetting your password feels like you should be logged out of any other locations where "you" are logged in, cause that is likely not-actually-you.